src/widgets/neuralWidget.py

```diff
@@ -333,26 +339,69 @@ def loadNeural(self, ca_file, sample_rate, start_frame, stop_frame, time_start,
+            trace.addPolygon(self.createPoly(t_values, y_values[chan, :]))
+            self.tracesCache[str(chan)] = trace
+
+    def createPoly(self, x_values, y_values):
+        if not (x_values.size == y_values.size == x_values.shape[0] == y_values.shape[0]):
+            raise ValueError("Arguments must be 1D NumPy arrays with same size")
+        size = x_values.size
+        poly = QPolygonF(size)
+        address = shiboken.getCppPointer(poly.data())[0]
+        buffer = (ctypes.c_double * 2 * size).from_address(address)
+        memory = np.frombuffer(buffer, np.float64)
+        memory[: (size - 1) * 2 + 1 : 2] = np.array(x_values, dtype=np.float64, copy=False)
+        memory[1 : (size - 1) * 2 + 2 : 2] = np.array(y_values, dtype=np.float64, copy=False)
+        
+        return poly
+
+    def putTracesIntoAGroup(self):
+        self.traces = QGraphicsItemGroup()
         pen = QPen()
         pen.setWidth(0)
-        trace = QPainterPath()
-        trace.reserve(self.stop_frame - self.start_frame + 1)
-        y = float(chan+0.5) + self.normalize(data[chan][self.start_frame])
-        trace.moveTo(self.time_start.float, y)
-        time_start_float = self.time_start.float
-
-        for ix in range(self.start_frame + 1, self.stop_frame):
-            t = (ix - self.start_frame)/self.sample_rate + time_start_float
-            val = self.normalize(data[chan][ix])
-            # Add a section to the trace path
-            y = float(chan+0.5) + val
-            trace.lineTo(t, y)
-        traceItem = QGraphicsPathItem(trace)
-        traceItem.setPen(pen)
-        self.traces.addToGroup(traceItem)
-
+        for k in list(self.tracesCache.keys()):
+            traceItem = QGraphicsPathItem(self.tracesCache[k])
+            traceItem.setPen(pen)
+            self.traces.addToGroup(traceItem)
+
+    def reorderTracesAndHeatmap(self, newOrder, showTraces, showHeatmap, showAnnotations):
+        _, counts = np.unique(newOrder, return_counts=True)
+        if np.any(counts > 1):
+            raise ValueError(f"Unique integer values should be there in newOrder array {newOrder}")
+        if newOrder.size != self.tracesOrder.size:
+            raise RuntimeError(f"Size of newOrder array and traces order should be the same.")
+        for item in self.items():
+            if isinstance(item, QGraphicsItemGroup):
+                self.removeItem(item)
+            elif isinstance(item, QGraphicsPixmapItem):
+                self.removeItem(item)
+        self.createHeatmap(newOrder)
+        offset = (self.tracesOrder[newOrder] - self.tracesOrder)
+        self.tracesOrder = self.tracesOrder + offset
+        for chan in list(self.tracesCache.keys()):
+            self.tracesCache[chan] = self.tracesCache[chan].translated(0., offset[int(chan)])
+        self.putTracesIntoAGroup()
+        self.addItem(self.traces)
+        self.setVisibility(showTraces, showHeatmap, showAnnotations)
```

Comment:
do we need to re-call setVisibility at the end of this function? Given that reorderTracesAndHeatmap is a function we would want plugin developers able to call, we should keep its number of required inputs as small as possible, either by setting default values of showTraces, showHeatmap, and showAnnotations, or just by not changing their values.